llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -2670,97 +2766,14 @@ bool X86DAGToDAGISel::matchAddressRecursively(SDValue N, X86ISelAddressMode &AM,
     break;
   }
   case ISD::ZERO_EXTEND: {
-    // Try to widen a zexted shift left to the same size as its use, so we can
-    // match the shift as a scale factor.
     if (AM.IndexReg.getNode() != nullptr || AM.Scale != 1)
```

Comment:
Can we remove the `AM.Scale != 1`? matchIndexRecursively needs to be able to accumulate scale